models/event_test.go

```diff
@@ -0,0 +1,159 @@
+package models
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestRemoveDuplicateEvents(t *testing.T) {
+	Convey("Given the stored filter output doesn't contain any events", t, func() {
+		current := &Filter{
+			ID:     "1234",
+			Events: []*Event{},
+		}
+
+		Convey("When filter a new filter provides no events", func() {
+			new := &Filter{
+				ID:     "1234",
+				Events: []*Event{},
+			}
+
+			Convey("Then the result should still be empty", func() {
+				new.RemoveDuplicateEvents(current)
+				So(len(new.Events), ShouldEqual, 0)
+			})
+		})
+
+		Convey("When filter a new filter provides events", func() {
+			new := &Filter{
+				Events: []*Event{
+					{
+						Type: "Event1",
+						Time: time.Now(),
+					},
+				},
+			}
+
+			Convey("Then all of them should be kept", func() {
+				new.RemoveDuplicateEvents(current)
+				So(len(new.Events), ShouldEqual, 1)
+			})
+		})
+
+	})
+
+	Convey("Given the stored filter output does contain events", t, func() {
+		e := &Event{
+			Type: "Event1",
+			Time: time.Now(),
+		}
+
+		current := &Filter{
+			Events: []*Event{e},
+		}
+
+		Convey("When filter a new filter provides no events", func() {
+			new := &Filter{
+				ID:     "1234",
+				Events: []*Event{},
+			}
+
+			Convey("Then the result should be empty", func() {
+				new.RemoveDuplicateEvents(current)
+				So(len(new.Events), ShouldEqual, 0)
+			})
+		})
+
+		Convey("When filter a new filter provides a new event", func() {
+			new := &Filter{
+				Events: []*Event{
+					{
+						Type: "Event2",
+						Time: time.Now(),
+					},
+				},
+			}
+
+			Convey("Then the new event should be kept", func() {
+				new.RemoveDuplicateEvents(current)
+				So(len(new.Events), ShouldEqual, 1)
```

Comment:
I would also assert that the event that remains is the right one if possible (ie Event2, and not Event1). It would probably make more sense if we had filters with multiple events though